### PR TITLE
Update to PMP R, W, and X bits

### PR DESCRIPTION
**Pull request by kersten1**
Making the change that Andrew suggested in https://github.com/riscv/riscv-isa-manual/pull/909

This should also close https://github.com/riscv/riscv-isa-manual/pull/908.

### CODE REVIEW

**Comment by aswaterman** [comment]
@kersten1 can you post a screenshot of what it looks like now?

**Comment by kersten1** [comment]
@aswaterman I will! Still figuring out why it failed.

**Comment by kersten1** [comment]
Screenshot 2024-04-30 at 3 34 10 PM

**Comment by aswaterman** [comment]
Can you work on improving the horizontal alignments?  E.g. the "4" is far to the right, and the cell contents are not centered.  I'm not sure what the right incantation is, but it's worth finding out.

Otherwise, lgtm.

File: src/images/bytefield/pmpcfg.adoc [hunk 1]
```diff
@@ -16,23 +16,23 @@
 (draw-box "1" {:span 5 :borders {}})
 (draw-box "0" {:span 5 :borders {}})
 
-(draw-box "L " {:span 2 :text-anchor "end" :borders{:top :border-unrelated :bottom :border-unrelated :left :border-unrelated}})
-(draw-box (text "(WARL)" {:font-weight "bold" :font-size 24}) {:span 3 :text-anchor "start" :borders{:top :border-unrelated :bottom :border-unrelated :right :border-unrelated}})
+(draw-box "L " {:span 3 :text-anchor "end" :borders{:top :border-unrelated :bottom :border-unrelated :left :border-unrelated}})
+(draw-box (text " " {:font-weight "bold" :font-size 24}) {:span 3 :text-anchor "start" :borders{:top :border-unrelated :bottom :border-unrelated :right :border-unrelated}})
```
Comment (kersten1) [inline src/images/bytefield/pmpcfg.adoc:20]:
```suggestion
(draw-box (text " " {:font-weight "bold" :font-size 24}) {:span 2 :text-anchor "start" :borders{:top :border-unrelated :bottom :border-unrelated :right :border-unrelated}})
```

File: src/images/bytefield/pmpcfg.adoc [hunk 2]
```diff
@@ -16,23 +16,23 @@
...
-(draw-box "W " {:span 2 :text-anchor "end" :borders{:top :border-unrelated :bottom :border-unrelated :left :border-unrelated}})
-(draw-box (text "(WARL)" {:font-weight "bold" :font-size 24}) {:span 3 :text-anchor "start" :borders{:top :border-unrelated :bottom :border-unrelated :right :border-unrelated}})
-(draw-box "R " {:span 2 :text-anchor "end" :borders{:top :border-unrelated :bottom :border-unrelated :left :border-unrelated}})
-(draw-box (text "(WARL)" {:font-weight "bold" :font-size 24}) {:span 3 :text-anchor "start" :borders{:top :border-unrelated :bottom :border-unrelated :right :border-unrelated}})
+(draw-box (text " " {:font-weight "bold" :font-size 24}) {:span 3 :text-anchor "start" :borders{:top :border-unrelated :bottom :border-unrelated :right :border-unrelated}})
+(draw-box "X " {:span 3 :text-anchor "end" :borders{:top :border-unrelated :bottom :border-unrelated :left :border-unrelated}})
+(draw-box (text " " {:font-weight "bold" :font-size 24}) {:span 3 :text-anchor "start" :borders{:top :border-unrelated :bottom :border-unrelated :right :border-unrelated}})
```
Comment (kersten1) [inline src/images/bytefield/pmpcfg.adoc:26]:
```suggestion
(draw-box (text " " {:font-weight "bold" :font-size 24}) {:span 2 :text-anchor "start" :borders{:top :border-unrelated :bottom :border-unrelated :right :border-unrelated}})
```

File: src/images/bytefield/pmpcfg.adoc [hunk 3]
```diff
@@ -16,23 +16,23 @@
...
+(draw-box "W " {:span 3 :text-anchor "end" :borders{:top :border-unrelated :bottom :border-unrelated :left :border-unrelated}})
+(draw-box (text " " {:font-weight "bold" :font-size 24}) {:span 3 :text-anchor "start" :borders{:top :border-unrelated :bottom :border-unrelated :right :border-unrelated}})
```
Comment (kersten1) [inline src/images/bytefield/pmpcfg.adoc:28]:
```suggestion
(draw-box (text " " {:font-weight "bold" :font-size 24}) {:span 2 :text-anchor "start" :borders{:top :border-unrelated :bottom :border-unrelated :right :border-unrelated}})
```

File: src/images/bytefield/pmpcfg.adoc [hunk 4]
```diff
@@ -16,23 +16,23 @@
...
+(draw-box "R " {:span 3 :text-anchor "end" :borders{:top :border-unrelated :bottom :border-unrelated :left :border-unrelated}})
+(draw-box (text " " {:font-weight "bold" :font-size 24}) {:span 3 :text-anchor "start" :borders{:top :border-unrelated :bottom :border-unrelated :right :border-unrelated}})
```
Comment (kersten1) [inline src/images/bytefield/pmpcfg.adoc:30]:
```suggestion
(draw-box (text " " {:font-weight "bold" :font-size 24}) {:span 2 :text-anchor "start" :borders{:top :border-unrelated :bottom :border-unrelated :right :border-unrelated}})
```

**Comment by kersten1** [comment]
This is what it should look like? 
Screenshot 2024-05-01 at 8 52 30 AM

**Comment by aswaterman** [comment]
What you posted in https://github.com/riscv/riscv-isa-manual/pull/1378#issuecomment-2087159177 was fine except for the misalignment of the contents of the cells (e.g. the "L" is all the way at the left of the cell instead of centered) and some of the cell headers (e.g. the "4" should be at the left of the cell boundary close to the "5" but is shifted over to the right).

File: src/smcntrpmf.adoc [hunk 5]
```diff
@@ -1,7 +1,7 @@
 [[smcntrpmf]]
 == "Smcntrpmf" Cycle and Instret Privilege Mode Filtering, Version 1.0.0
 
-[[intro]]
+[[introsmcentrpmf]]
```
Comment (kersten1) [inline src/smcntrpmf.adoc:4]:
```suggestion
[[intro]]
```

File: src/images/bytefield/pmpcfg.adoc [hunk 6]
```diff
@@ -16,23 +16,23 @@
 (draw-box "1" {:span 5 :borders {}})
 (draw-box "0" {:span 5 :borders {}})
 
-(draw-box "L " {:span 2 :text-anchor "end" :borders{:top :border-unrelated :bottom :border-unrelated :left :border-unrelated}})
-(draw-box (text "(WARL)" {:font-weight "bold" :font-size 24}) {:span 3 :text-anchor "start" :borders{:top :border-unrelated :bottom :border-unrelated :right :border-unrelated}})
+(draw-box "L " {:span 3 :text-anchor "end" :borders{:top :border-unrelated :bottom :border-unrelated :left :border-unrelated}})
```
Comment (kersten1) [inline src/images/bytefield/pmpcfg.adoc:19]:
```suggestion
(draw-box "L" {:span 3 :text-anchor "end" :borders{:top :border-unrelated :bottom :border-unrelated :left :border-unrelated}})
```

File: src/images/bytefield/pmpcfg.adoc [hunk 7]
```diff
@@ -16,23 +16,23 @@
 (draw-box "1" {:span 5 :borders {}})
 (draw-box "0" {:span 5 :borders {}})
 
-(draw-box "L " {:span 2 :text-anchor "end" :borders{:top :border-unrelated :bottom :border-unrelated :left :border-unrelated}})
-(draw-box (text "(WARL)" {:font-weight "bold" :font-size 24}) {:span 3 :text-anchor "start" :borders{:top :border-unrelated :bottom :border-unrelated :right :border-unrelated}})
+(draw-box "L" {:span 3 :text-anchor "end" :borders{:top :border-unrelated :bottom :border-unrelated :left :border-unrelated}})
+(draw-box (text " " {:font-weight "bold" :font-size 24}) {:span 2 :text-anchor "start" :borders{:top :border-unrelated :bottom :border-unrelated :right :border-unrelated}})
 (draw-box "0 " {:span 3 :text-anchor "end" :borders{:top :border-unrelated :bottom :border-unrelated :left :border-unrelated}})
```
Comment (kersten1) [inline src/images/bytefield/pmpcfg.adoc:21]:
```suggestion
(draw-box "0" {:span 3 :text-anchor "end" :borders{:top :border-unrelated :bottom :border-unrelated :left :border-unrelated}})
```

File: src/images/bytefield/pmpcfg.adoc [hunk 8]
```diff
@@ -16,23 +16,23 @@
...
-(draw-box "0 " {:span 3 :text-anchor "end" :borders{:top :border-unrelated :bottom :border-unrelated :left :border-unrelated}})
-(draw-box (text "(WARL)" {:font-weight "bold" :font-size 24}) {:span 3 :text-anchor "start" :borders{:top :border-unrelated :bottom :border-unrelated :right :border-unrelated}})
+(draw-box "L" {:span 3 :text-anchor "end" :borders{:top :border-unrelated :bottom :border-unrelated :left :border-unrelated}})
+(draw-box (text " " {:font-weight "bold" :font-size 24}) {:span 2 :text-anchor "start" :borders{:top :border-unrelated :bottom :border-unrelated :right :border-unrelated}})
+(draw-box "0" {:span 3 :text-anchor "end" :borders{:top :border-unrelated :bottom :border-unrelated :left :border-unrelated}})
+(draw-box (text " " {:font-weight "bold" :font-size 24}) {:span 3 :text-anchor "start" :borders{:top :border-unrelated :bottom :border-unrelated :right :border-unrelated}})
 (draw-box "A " {:span 3 :text-anchor "end" :borders{:top :border-unrelated :bottom :border-unrelated :left :border-unrelated}})
```
Comment (kersten1) [inline src/images/bytefield/pmpcfg.adoc:23]:
```suggestion
(draw-box "A" {:span 3 :text-anchor "end" :borders{:top :border-unrelated :bottom :border-unrelated :left :border-unrelated}})
```

File: src/images/bytefield/pmpcfg.adoc [hunk 9]
```diff
@@ -16,23 +16,23 @@
...
+(draw-box "L" {:span 3 :text-anchor "end" :borders{:top :border-unrelated :bottom :border-unrelated :left :border-unrelated}})
+(draw-box (text " " {:font-weight "bold" :font-size 24}) {:span 2 :text-anchor "start" :borders{:top :border-unrelated :bottom :border-unrelated :right :border-unrelated}})
+(draw-box "0" {:span 3 :text-anchor "end" :borders{:top :border-unrelated :bottom :border-unrelated :left :border-unrelated}})
+(draw-box (text " " {:font-weight "bold" :font-size 24}) {:span 3 :text-anchor "start" :borders{:top :border-unrelated :bottom :border-unrelated :right :border-unrelated}})
+(draw-box "A" {:span 3 :text-anchor "end" :borders{:top :border-unrelated :bottom :border-unrelated :left :border-unrelated}})
+(draw-box (text " " {:font-weight "bold" :font-size 24}) {:span 3 :text-anchor "start" :borders{:top :border-unrelated :bottom :border-unrelated :right :border-unrelated}})
+(draw-box "X " {:span 3 :text-anchor "end" :borders{:top :border-unrelated :bottom :border-unrelated :left :border-unrelated}})
```
Comment (kersten1) [inline src/images/bytefield/pmpcfg.adoc:25]:
```suggestion
(draw-box "X" {:span 3 :text-anchor "end" :borders{:top :border-unrelated :bottom :border-unrelated :left :border-unrelated}})
```

File: src/images/bytefield/pmpcfg.adoc [hunk 10]
```diff
@@ -16,23 +16,23 @@
...
+(draw-box "0" {:span 3 :text-anchor "end" :borders{:top :border-unrelated :bottom :border-unrelated :left :border-unrelated}})
+(draw-box (text " " {:font-weight "bold" :font-size 24}) {:span 3 :text-anchor "start" :borders{:top :border-unrelated :bottom :border-unrelated :right :border-unrelated}})
+(draw-box "A" {:span 3 :text-anchor "end" :borders{:top :border-unrelated :bottom :border-unrelated :left :border-unrelated}})
+(draw-box (text " " {:font-weight "bold" :font-size 24}) {:span 3 :text-anchor "start" :borders{:top :border-unrelated :bottom :border-unrelated :right :border-unrelated}})
+(draw-box "X" {:span 3 :text-anchor "end" :borders{:top :border-unrelated :bottom :border-unrelated :left :border-unrelated}})
+(draw-box (text " " {:font-weight "bold" :font-size 24}) {:span 2 :text-anchor "start" :borders{:top :border-unrelated :bottom :border-unrelated :right :border-unrelated}})
+(draw-box "W " {:span 3 :text-anchor "end" :borders{:top :border-unrelated :bottom :border-unrelated :left :border-unrelated}})
```
Comment (kersten1) [inline src/images/bytefield/pmpcfg.adoc:27]:
```suggestion
(draw-box "W" {:span 3 :text-anchor "end" :borders{:top :border-unrelated :bottom :border-unrelated :left :border-unrelated}})
```

File: src/images/bytefield/pmpcfg.adoc [hunk 11]
```diff
@@ -16,23 +16,23 @@
...
+(draw-box "A" {:span 3 :text-anchor "end" :borders{:top :border-unrelated :bottom :border-unrelated :left :border-unrelated}})
+(draw-box (text " " {:font-weight "bold" :font-size 24}) {:span 3 :text-anchor "start" :borders{:top :border-unrelated :bottom :border-unrelated :right :border-unrelated}})
+(draw-box "X" {:span 3 :text-anchor "end" :borders{:top :border-unrelated :bottom :border-unrelated :left :border-unrelated}})
+(draw-box (text " " {:font-weight "bold" :font-size 24}) {:span 2 :text-anchor "start" :borders{:top :border-unrelated :bottom :border-unrelated :right :border-unrelated}})
+(draw-box "W" {:span 3 :text-anchor "end" :borders{:top :border-unrelated :bottom :border-unrelated :left :border-unrelated}})
+(draw-box (text " " {:font-weight "bold" :font-size 24}) {:span 2 :text-anchor "start" :borders{:top :border-unrelated :bottom :border-unrelated :right :border-unrelated}})
+(draw-box "R " {:span 3 :text-anchor "end" :borders{:top :border-unrelated :bottom :border-unrelated :left :border-unrelated}})
```
Comment (kersten1) [inline src/images/bytefield/pmpcfg.adoc:29]:
```suggestion
(draw-box "R" {:span 3 :text-anchor "end" :borders{:top :border-unrelated :bottom :border-unrelated :left :border-unrelated}})
```

**Comment by kersten1** [comment]
Sorry for the delay with this one but I learned a lot about bytefield images! 
Screenshot 2024-05-02 at 3 53 04 PM

**Comment by aswaterman** [comment]
Cool, thank you!

**Pull request merged**